### PR TITLE
feat: Switch schedule_viewer.py map from Google to Leaflet

### PR DESCRIPTION
**Pull request by losvedir**
Since the change to the Google Maps API that now requires an API key with billing enabled, we can't easily use this. Leaflet with OSM is a great alternative, and the Leaflet JS API is fairly similar to Google Maps, so it was fairly straightforward to switch to it. 

It should be nearly functionally identical. There was a bit of behavior in the code around selected stop text that I didn't keep since Leaflet doesn't have a `marker.getText()` and `marker.setText()` equivalent. I could potentially put that in a `Popup` but then I'd have to bind Popups to all the markers. I never used this functionality before, and with Google Maps broken now I couldn't test it out, so I'm not really sure what we're missing. The page works in all the ways I've used it.

To try it out, it should be as easy as cloning this repo and checking out this branch, and running `python2.7 schedule_viewer.py --feed=/path/to/build/current/`.

Screen Shot 2021-04-09 at 3 37 07 PM

### CODE REVIEW

File: gtfsscheduleviewer/files/index.js [hunk 1]
```diff
@@ -11,25 +10,18 @@ function load(config) {
...
-  var mapOptions = {
-    zoom: 13,
-    center: config.bounds.getCenter(),
-    mapTypeId : google.maps.MapTypeId.ROADMAP,
-    scaleControl: true
-  };
-  map = new google.maps.Map(map_dom, mapOptions);
+
+  map = L.map('map')
+  L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
```
Comment (paulswartz) [inline gtfsscheduleviewer/files/index.js]: Can you use our tiles? `https://cdn.mbta.com/osm_tiles/{z}/{x}/{y}.png` (attribution can stay the same).

**Review by paulswartz — APPROVED** [review]
one small note, otherwise 🍰

**Review by jfabi — COMMENTED** [review]
>There was a bit of behavior in the code around selected stop text that I didn't keep since Leaflet doesn't have a `marker.getText()` and `marker.setText()` equivalent. I could potentially put that in a Popup but then I'd have to bind Popups to all the markers. I never used this functionality before, and with Google Maps broken now I couldn't test it out, so I'm not really sure what we're missing.

I don't know if this is the exact behavior you're referring to, but there used to be these labels next to each stop marker showing the arrival/departure time for the respective stop. I think they're pretty handy because sometimes the popups won't actually show the _selected_ trip's time if there are too many trips around the time you select in the UI.

It looks like you could probably use [tooltips in Leaflet (which bind to markers)](https://leafletjs.com/reference-1.7.1.html#tooltip) to recreate the old behavior, or you could add the time to the existing markers (maybe in a bigger/bolder type at the beginning of the body?).

(Screenshots from previous pull requests.)

image
image

A more minor qualm, really with the default OSM style, is that a lot more roads blend in with bus routes with little color contrast.

Screenshot 2021-04-09 at 18 16 44
Screenshot 2021-04-09 at 18 14 53

I tried using another basemap style that's free/doesn't require a key, and it seemed maybe marginally better, but I'm not sure if it's enough to warrant a change—see below. Let me know if you'd prefer it as a likely future user, though, and I can get you the details.

Screenshot 2021-04-12 at 11 24 34
Screenshot 2021-04-12 at 11 21 02

**Comment by losvedir** [comment]
@jfabi - I updated the tiles to use the ones we actually serve, and it clashes less. I also updated to use permanent tool-tips to display the times. Let me know what you think.

(I amended the commit rather than pushing up a fixup, so you'll have to `fetch` and then `reset --hard origin/gjd-maps-to-leaflet`.)

Screen Shot 2021-04-12 at 1 11 49 PM

**Review by paulswartz — APPROVED** [review]
🍰

**Review by jfabi — APPROVED** [review]
Excellent work! 🎆

**Pull request merged**